packages/backend-function/src/function_env_type_generator.ts

```diff
@@ -2,58 +2,61 @@ import fs from 'fs';
 import { staticEnvironmentVariables } from './static_env_types.js';
 import path from 'path';
 import os from 'os';
-import { FunctionProps } from './factory.js';
 
 /**
- * Generates a type definition file for environment variables
+ * Generates a typed process.env shim for environment variables
  */
 export class FunctionEnvironmentTypeGenerator {
   private typeDefFilePath: string;
-  private dynamicEnvironmentVariables: string[] = [];
 
   /**
-   * Initialize type definition file name and location
+   * Initialize typed process.env shim file name and location
    */
   constructor(
     functionName: string,
-    functionEnvironmentVariables: FunctionProps['environment']
+    private functionEnvironmentVariables: string[]
   ) {
     this.typeDefFilePath = `${process.cwd()}/.amplify/function-env/${functionName}.ts`;
-
-    for (const envName in functionEnvironmentVariables) {
-      this.dynamicEnvironmentVariables.push(envName);
-    }
   }
 
   /**
-   * Generate a type definition file
+   * Generate a typed process.env shim
    */
-  generateTypeDefFile() {
+  generateTypedProcessEnvShim() {
+    const lambdaEnvVarTypeName = 'LambdaProvidedEnvVars';
+    const definedEnvVarTypeName = 'DefinedEnvVars';
```

Comment:
nit: since we're currently only using these values in one place, I think it's more readable to just include the value in the string

Reply:
We're using them both in the type assertion of `process.env` and where we declare each type, unless you mean one place as only this file?